File: clients/typescript/src/models/IngestMedia.ts
/* tslint:disable */
/* eslint-disable */

import { mapValues } from '../runtime';

export interface IngestMedia {

    caption?: string | null;

    customerId?: string | null;

    data?: string | null;

    metadata?: { [key: string]: any; } | null;

    stripExif?: boolean;

    tags?: Array<string> | null;

    url?: string | null;
}

export function instanceOfIngestMedia(value: object): value is IngestMedia {
    return true;
}

export function IngestMediaFromJSON(json: any): IngestMedia {
    return IngestMediaFromJSONTyped(json, false);
}

export function IngestMediaFromJSONTyped(json: any, ignoreDiscriminator: boolean): IngestMedia {
    if (json == null) {
        return json;
    }
    return {

        'caption': json['caption'] === undefined ? undefined : json['caption'] === null ? null : json['caption'],
        'customerId': json['customer_id'] === undefined ? undefined : json['customer_id'] === null ? null : json['customer_id'],
        'data': json['data'] === undefined ? undefined : json['data'] === null ? null : json['data'],
        'metadata': json['metadata'] === undefined ? undefined : json['metadata'] === null ? null : json['metadata'],
        'stripExif': json['strip_exif'] == null ? undefined : json['strip_exif'],
        'tags': json['tags'] === undefined ? undefined : json['tags'] === null ? null : json['tags'],
        'url': json['url'] === undefined ? undefined : json['url'] === null ? null : json['url'],
    };
}

export function IngestMediaToJSON(json: any): IngestMedia {
    return IngestMediaToJSONTyped(json, false);
}

export function IngestMediaToJSONTyped(value?: IngestMedia | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {

        'caption': value['caption'],
        'customer_id': value['customerId'],
        'data': value['data'],
        'metadata': value['metadata'],
        'strip_exif': value['stripExif'],
        'tags': value['tags'],
        'url': value['url'],
    };
}

File: clients/rust/src/models/ingest_media.rs
use crate::models;
use serde::{Deserialize, Serialize};

#[derive(Clone, Default, Debug, PartialEq, Serialize, Deserialize)]
pub struct IngestMedia {
    #[serde(rename = "caption", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub caption: Option<Option<String>>,
    #[serde(rename = "customer_id", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub customer_id: Option<Option<String>>,
    #[serde(rename = "data", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub data: Option<Option<String>>,
    #[serde(rename = "metadata", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub metadata: Option<Option<std::collections::HashMap<String, serde_json::Value>>>,
    #[serde(rename = "strip_exif", skip_serializing_if = "Option::is_none")]
    pub strip_exif: Option<bool>,
    #[serde(rename = "tags", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub tags: Option<Option<Vec<String>>>,
    #[serde(rename = "url", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub url: Option<Option<String>>,
}

impl IngestMedia {
    pub fn new() -> IngestMedia {
        IngestMedia {
            caption: None,
            customer_id: None,
            data: None,
            metadata: None,
            strip_exif: None,
            tags: None,
            url: None,
        }
    }
}

